dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/common/logging/LogUtils.java

```diff
@@ -264,27 +269,46 @@ protected static Logger createLogger(final Class<?> cls,
                 try {
                     b = BundleUtils.getBundle(cls, bundleName);
                 } catch (MissingResourceException rex) {
-                    //ignore
                 }
             }
             if (b != null) {
                 b.getLocale();
             }
 
+            Class<?> wsLoggerClass = null;
+            try {
+                wsLoggerClass = Class.forName("com.ibm.ws.logging.internal.WsLogger");
+            } catch (ClassNotFoundException e) {
+                // ignore class not found
+                // it won't match loggerClass anyway
+            }
+            // Liberty change begin: Source Class is added as parameter
+            // Default logger of Liberty is WsLogger. 
+            // CXFLogger is inherited from this WsLogger class to preserve same behavior
+            // except the one that we aimed to change. Which is to add class name to log records
             if (loggerClass != null) {
+                Constructor<?> cns = null;
                 try {
-                    Constructor<?> cns = loggerClass.getConstructor(String.class, String.class);
-                    if (name == null) {
+                    // Liberty change begin: Replace WsLogger with CXFLogger
+                    if (loggerClass.equals(wsLoggerClass) || loggerClass.equals(CXFLogger.class)) {
+                        backupLoggerClass(CXFLogger.class);
                         try {
-                            return (Logger) cns.newInstance(loggerName, bundleName);
+                            cns = CXFLogger.class.getConstructor(String.class, String.class, Class.class);
+                            return (CXFLogger) cns.newInstance(loggerName, bundleName, cls);
                         } catch (InvocationTargetException ite) {
                             if (ite.getTargetException() instanceof MissingResourceException) {
-                                return (Logger) cns.newInstance(loggerName, null);
+                                return (CXFLogger) cns.newInstance(loggerName, null, cls);
```

Comment:
Is no exception is possible on this line?  If an exception is possible, then we wouldn't want to allow it. We'd want to fall back again to the original logger, that way no matter what exception occurs, we always just fall back to the original logger.